From 6edaaa4d1823518a97fb4cc3004d110b1046f742 Mon Sep 17 00:00:00 2001
From: Jonathan Kew <jfkthame@googlemail.com>
Date: Tue, 28 Apr 2020 16:50:49 +0000
Subject: [PATCH] Set name_mapping to NULL after freeing

To avoid risk of double-free.

Fixes #237.
---
 src/fcfreetype.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/fcfreetype.c b/src/fcfreetype.c
index cd8fa43..bfbd5d7 100644
--- a/src/fcfreetype.c
+++ b/src/fcfreetype.c
@@ -1627,7 +1627,10 @@ FcFreeTypeQueryFaceInternal (const FT_Face  face,
 	}
     }
     if (!nm_share)
+    {
 	free (name_mapping);
+	name_mapping = NULL;
+    }
 
     if (!nfamily && face->family_name &&
 	FcStrCmpIgnoreBlanksAndCase ((FcChar8 *) face->family_name, (FcChar8 *) "") != 0)
-- 
2.26.2

